djgentelella/tests/API_test.py

```diff
@@ -811,6 +860,83 @@ def test_handle_missing_related_permissions(self):
             permission_related = PermissionRelated.objects.filter(main_permission=main_permission)
             self.assertFalse(permission_related.exists())
 
-    # You can add more test methods for other scenarios as needed
 
 
+
+# Get an specific permission to the API
+#this test ensures that the API is able to correctly retrieve the details
+# of a specific PermissionRelated object and that the API response is consistent with
+# the data stored in the database for that object.
+class PermissionRelatedAPITests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.permission_related = PermissionRelated.objects.create(
+            main_permission_id=110,  # Reemplaza con el ID de un objeto existente
+            # Otras propiedades necesarias
+        )
+
+    def test_retrieve_permission_related_detail(self):
+        url = reverse('api_related_permissions_detail', kwargs={'permission_id': self.permission_related.main_permission_id})
+        response = self.client.get(url)
+
+        # Verifica que la respuesta tenga un código de estado 200 (OK)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Verifica que los datos recuperados sean correctos
+        self.assertEqual(response.data['main_permission'], self.permission_related.main_permission_id)
+        # Agrega más verificaciones según tus campos y datos específicos
+
+    # Otras pruebas de ser necesario
+
+
+ #non existing permission using URL
+#this test ensures that the API correctly handles the request for details
+# of a PermissionRelated object that does not exist in the database,
+# returning a 404 status code to indicate that the resource is not found.
+class PermissionRelatedAPITest(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_retrieve_nonexistent_permission_related_detail(self):
+        # Define un 'permission_id' que no existe en la base de datos
+        nonexistent_permission_id = 9999
+
+        url = reverse('api_related_permissions_detail', kwargs={'permission_id': nonexistent_permission_id})
+        response = self.client.get(url)
+
+        # Verifica que la respuesta tenga un código de estado 404 (Not Found)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
```

Comment:
This test can be allocated in the above class `PermissionRelatedAPITests` there is not need for isolating this in another class.